sw/device/mask_rom/boot.md

```diff
@@ -209,63 +446,130 @@ crt_init() {
-  // - Pinmux / Padctrl
-  // - Entropy Configuration
-  // - Anything Else?
+Extensibility: None. A ROM_EXT can disable the use of a key, but cannot add new
+valid root key IDs. There will be a fixed set of root key IDs as part of a given
+Mask ROM version.
 
-  // This is explicitly a separate step from locking the key manager inputs. In
-  // particular, this depends on the ROM_EXT's choices, whereas the key manager
-  // inputs getting locked is not something the ROM_EXT can choose.
-}
-```
+### Boot Policy Structure
+
+Accessed by:
+
+*   Mask ROM (to choose ROM_EXT, and during bootstrapping to bless new ROM_EXT).
+*   ROM_EXT (during firmware update).
+
+Needs to contain:
+
+*   Identifier (so we know we're reading the right thing)
+
+    This also acts like a version number, because the Mask ROM code that parses
+    the boot policy can never be updated. Conversely, any changes to this
+    structure require new Mask ROM parsing code, which should be denoted with a
+    new identifier.
+
+*   Which ROM_EXT slot should be chosen first (2.b, 2.c.i).
+*   What to do if ROM_EXT does not validate (2.c.ii, 2.c.iii):
+    *   Try Alternate ROM_EXT; or
+    *   Fail to Boot
+*   What to do if ROM_EXT validates successfully, just before jumping to ROM_EXT:
+    *   Do nothing; or
+    *   Set current ROM_EXT as Primary if not already.
+*   Checksum (of everything else).
+
+Stored in: Flash (Info Partition)
+
+Extensibility: None. This info only controls the actions of the Mask ROM. You
+cannot add functionality to the Mask ROM of a given chip, so there's no way to
+add other information to this structure.
+
+The in-RAM representation also contains the boot reason (reset manager), because
+it needs to be checked most times the boot policy is also read.
+
+### ROM_EXT Manifest Structure
+
+Accessed by:
+
+*   Mask ROM (to validate ROM_EXT)
+*   BL0 Kernel (during firmware update).
+
+Needs to contain:
+
+*   Unsigned Area:
+    *   Identifier (so we know we're reading the right thing)
+        This also acts as a ROM_EXT manifest version (ie, the version of the
+        layout of the header).
+
+    *   Signature
+*   Signed Area:
+    *   Image Length
+    *   Entry Point
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   ROM_EXT PMP Region Information
+        **Open Q:** Is this fixed or programmable? Field not needed if fixed.
+
+    *   Key Selection Area
+        *   Public Key
+
+    *   Software Binding Properties:
+        *   ROM_EXT Version
+            This is the version of the image contained in the ROM_EXT.
+
+        *   Usage Constraints
+        *   Peripheral Lockdown Information
+    *   Read-only Extension Area:
+        *   Several words for ROM_EXT/BL0 use only.
+            (Interpretation governed by ROM_EXT/BL0 Version).
+            **Open Q:** Does this satisfy the previous usecases?
+
+            This is likely to be a fixed number of `(pointer, checksum)` pairs,
```

Comment:
Also, consider adding some alignment (offset, size) requirements if it helps to simplify the implementation (e.g. 32B aligned offsets, sizes). In some cases it may even make sense to make things page aligned.